xarray/core/variable.py

```diff
@@ -159,12 +160,7 @@ def as_compatible_data(data, fastpath=False):
     if isinstance(data, Variable):
         return data.data
 
-    # add a custom fast-path for dask.array to avoid expensive checks for the
-    # dtype attribute
-    if isinstance(data, dask_array_type):
-        return data
-
-    if isinstance(data, pd.Index):
+    if isinstance(data, SUPPORT_ARRAY_TYPES):
```

Comment:
I don't think this is quite right -- we still cast certain types of ndarray arguments at the end of this function (e.g., with `_possibly_convert_objects`). I'm somewhat surprised tests aren't failing.

Reply:
`SUPPORT_ARRAY_TYPES` does not include `np.ndarray` (I changed from the previous commit), and some type-castings for `np.ndarray` in below are still working here.

(But maybe my naming `SUPPORT_ARRAY_TYPES` is not good as `np.ndarray` is also supported.)